static/app/views/performance/landing/index.tsx

```diff
@@ -199,91 +203,109 @@ export function PerformanceLanding(props: Props) {
-                                <SearchBar
-                                  searchSource="performance_landing"
-                                  organization={organization}
-                                  projectIds={eventView.project}
-                                  query={searchQuery}
-                                  fields={generateAggregateFields(
-                                    organization,
-                                    [...eventView.fields, {field: 'tps()'}],
-                                    ['epm()', 'eps()']
-                                  )}
-                                  onSearch={(query: string) =>
-                                    handleSearch(query, metricSettingState ?? undefined)
-                                  }
-                                  maxQueryLength={MAX_QUERY_LENGTH}
-                                />
-                              );
-                            }}
-                          </MEPConsumer>
-                          <MetricsEventsDropdown />
-                        </SearchFilterContainer>
-                        {initiallyLoaded ? (
-                          <TeamKeyTransactionManager.Provider
+                      forceTransactions={metricsDataSide.forceTransactionsOnly}
+                    >
+                      <MetricsDataSwitcherAlert
+                        organization={organization}
+                        eventView={eventView}
+                        projects={projects}
+                        location={location}
+                        router={props.router}
+                        {...metricsDataSide}
+                      />
+                      {!metricsDataSide.shouldWarnIncompatibleSDK &&
+                        !metricsDataSide.shouldNotifyUnnamedTransactions && (
+                          <DynamicSamplingMetricsAccuracyAlert
                             organization={organization}
-                            teams={teams}
-                            selectedTeams={['myteams']}
-                            selectedProjects={eventView.project.map(String)}
-                          >
-                            <GenericQueryBatcher>
-                              <ViewComponent {...props} />
-                            </GenericQueryBatcher>
-                          </TeamKeyTransactionManager.Provider>
-                        ) : (
-                          <LoadingIndicator />
+                            selectedProject={
+                              selectedProjects.length === 1 &&
+                              String(selectedProjects[0]) !== String(ALL_ACCESS_PROJECTS) // -1 means that all projects are selected
+                                ? projects.find(
+                                    project =>
+                                      String(project.id) === String(selectedProjects[0])
+                                  )
+                                : undefined
+                            }
```

Comment:
I was going to suggest moving this out to a util, but then realized we do this a lot, so I put up a PR to add this functionality (including My Projects behaviour, which we overlook a lot 😓) into `EventView` here: https://github.com/getsentry/sentry/pull/39563